crates/typst/src/layout/length.rs

```diff
@@ -81,12 +81,15 @@ impl Length {
         if self.em == Em::zero() {
             return Ok(());
         }
-        Err(eco_format!(
+
+        bail!(
+            span,
             "cannot convert a length with non-zero em units (`{}`) to {unit}",
-            self.repr()
-        ))
-        .hint(eco_format!("use `length.abs.{unit}()` instead to ignore its em component"))
-        .at(span)
+            self.repr();
+            hint: "use `length.abs.{unit}()` instead to ignore its em component";
+            hint: "or use `length.to-absolute()` to resolve its em component \
+                   (requires a context)"
```

Comment:
```suggestion
                   (requires context)"
```
maybe we should also flip the order of the hints? ignoring is generally not as advisable.